### PR TITLE
docs: improve retry configuration documentation

### PR DESCRIPTION
**Pull request by dmbrooke**

### CODE REVIEW

File: src/environment.ts [hunk 1]
```diff
@@ -11,26 +11,51 @@ export enum PlatformEnvironment {
 export const DEFAULT_ENVIRONMENT = PlatformEnvironment.Prod as const;
 export const DEFAULT_REGION = Region.US as const;
 export const DEFAULT_RETRY_AFTER = 5000;
-export const DEFAULT_MAX_RETRIES = 50;
+export const DEFAULT_MAX_RETRIES = 10;
 export const DEFAULT_TIME_MULTIPLE = 2;
 
-export type PlatformUrlOptions = {
+export type PlatformOptions = {
```
Comment (dmbrooke) [inline src/environment.ts:17]: @y-lakhdar brought up a good point that renaming this is potentially a breaking change. Even though I'm not sure if there's much of a reason for implementers to actually import this type, it's still publicly exported.

That being said, does this still fall under "minor release" so long as we clarify that this is a breaking change?

Comment (louis-bompart) [reply src/environment.ts:17]: breaking is breaking. But you can:
```suggestion
@deprecated use @link {PlatformOptions}
export type PlatformUrlOptions = PlatformOptions;
export type PlatformOptions = {
```

File: README.md [hunk 2]
```diff
@@ -53,6 +53,41 @@ See more examples in the `./samples` folder.
...
+You may configure the exponential backoff that will be applied to all outgoing requests. To do so, specify a `PlatformOptions` object when creating either a `CatalogSource` or `PushSource` object:
+
+```js
+const catalogSource = new CatalogSource(
+  'my_api_key',
+  'my_coveo_organization_id',
+  {maxRetries: 10, retryAfter: 2000, timeMultiple: 3}
+);
+const pushSource = new PushSource(
+  'my_api_other_key',
```
Comment (mrrajamanickam-coveo) [inline README.md:83]: nit:
```suggestion
  'my_other_api_key',
```

File: README.md [hunk 3]
```diff
@@ -53,6 +53,41 @@ See more examples in the `./samples` folder.
...
+  'my_other_coveo_organization_id',
+  {maxRetries: 3, retryAfter: 600000}
+);
+```
+
+By default, requests will retry a maximum of 10 times, waiting 5 seconds after the first attempt, with a time multiple of 2 (which will equate to a maximum execution time of roughly 1.5 hours).
```
Comment (mrrajamanickam-coveo) [inline README.md:89]: Was this text present in the Python SDK docs?

Comment (dmbrooke) [reply README.md:89]: Ah, this must have been lost in translation from MD to RST. I'll add this paragraph in the python SDK PR 🙂

File: src/environment.ts (hunk 1 above)
Comment (dmbrooke) [reply src/environment.ts:22]: Great suggestion! Applied ✅

**Review by mrrajamanickam-coveo — APPROVED** [review]

**Review by olamothe — APPROVED** [review]

**Pull request merged**